File: src/main/java/ru/timakden/chesspuzzle/King.java
package ru.timakden.chesspuzzle;

/**
 * Король, перемещается на любое соседнее поле в любом направлении.
 */
class King extends ChessPiece {
	King(int row, int column) {
		super(row, column);
	}

	@Override
	public String toString() {
		return "K";
	}

	public boolean canAttackAnotherChessPiece(ChessPiece anotherChessPiece) {
		int row = getRow();
		int column = getColumn();
		int anotherRow = anotherChessPiece.getRow();
		int anotherColumn = anotherChessPiece.getColumn();

		return ((Math.abs(row - anotherRow) == 1) && (Math.abs(column - anotherColumn) == 1)) ||
		       ((row == anotherRow) && (Math.abs(column - anotherColumn) == 1)) ||
		       ((column == anotherColumn) && (Math.abs(row - anotherRow) == 1));
	}
}

File: src/main/kotlin/ru/timakden/chesspuzzle/King.kt
package ru.timakden.chesspuzzle

/**
 * Король, перемещается на любое соседнее поле в любом направлении.
 */
class King(row: Int, column: Int) : ChessPiece(row, column) {
    override fun toString() = "K"

    override fun canAttackAnotherPiece(anotherPiece: ChessPiece): Boolean {
        return Math.abs(row - anotherPiece.row) == 1 && Math.abs(column - anotherPiece.column) == 1 ||
                row == anotherPiece.row && Math.abs(column - anotherPiece.column) == 1 ||
                column == anotherPiece.column && Math.abs(row - anotherPiece.row) == 1
    }
}
